add test file for pkg/kubectl/cmd/rollout/rollout_pause.go file

WanLinghao opened the pull request:
new:   pkg/kubectl/cmd/rollout/rollout_pause_test.go
        modified: pkg/kubectl/cmd/rollout/BUILD
<!--  Thanks for sending a pull request!  Here are some tips for you:
1. If this is your first time, read our contributor guidelines https://git.k8s.io/community/contributors/devel/pull-requests.md#the-pr-submit-process and developer guide https://git.k8s.io/community/contributors/devel/development.md#development-guide
2. If you want *faster* PR reviews, read how: https://git.k8s.io/community/contributors/devel/pull-requests.md#best-practices-for-faster-reviews
3. Follow the instructions for writing a release note: https://git.k8s.io/community/contributors/devel/pull-requests.md#write-release-notes-if-needed
-->

**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:
<!--  Steps to write your release note:
1. Use the release-note-* labels to set the release note state (if you have access)
2. Enter your extended release note in the below block; leaving it blank means using the PR title as the release note. If no release note is required, just write `NONE`.
-->
```release-note
```

xiangpengzhao commented:
/release-note-none
/ok-to-test

please amend your commit message to be more meaningful, also remove the Chinese characters :)

WanLinghao commented:
/retest

WanLinghao commented:
/retest

WanLinghao commented:
/retest

WanLinghao commented:
@xiangpengzhao thanks for the reply.I am in trouble with the visibility in BUILD file.can you offer any help or clue?

WanLinghao commented:
@lavalamp @smarterclayton  please take a look :)

xiangpengzhao commented:
@WanLinghao sorry for the delay. Seems like you have already fixed the BUILD problem :+1: 

In the future, you can try to run `hack/update-bazel.sh` if you change the import of packages in your PR.

WanLinghao commented:
@kargakis @janetkuo PTAL

kargakis commented:
/lgtm

WanLinghao commented:
PTAL,thanks  @dchen1107  @pwittrock

WanLinghao commented:
/retest

WanLinghao commented:
@dchen1107 PTAL

janetkuo commented on pkg/kubectl/cmd/rollout/rollout_pause_test.go:
```diff
@@ -0,0 +1,115 @@
+		RESTClient: &fake.RESTClient{
+			APIRegistry:          api.Registry,
+			NegotiatedSerializer: ns,
+			Client: fake.CreateHTTPClient(func(req *http.Request) (*http.Response, error) {
```
Would you also add a switch-case on `req.URL.Path` and `req.Method` like the other unit tests do, and return error when the request is unexpected?

janetkuo commented:
/remove-lifecycle rotten

WanLinghao committed:
add UT test for rollout_pause.go file
	modified:   pkg/kubectl/cmd/rollout/BUILD
	new file:   pkg/kubectl/cmd/rollout/rollout_pause_test.go
	modified:   build/visible_to/BUILD

WanLinghao commented:
/test pull-kubernetes-bazel-test

WanLinghao commented:
/retest

WanLinghao commented:
@janetkuo  Sorry, I am in trouble with this visibility things, could tell me how to handle this?

WanLinghao commented:
/retest

WanLinghao commented:
/retest

WanLinghao commented:
/retest

WanLinghao commented:
@janetkuo this patch is ready

WanLinghao commented:
@janetkuo PTAL Thanks

janetkuo commented:
/lgtm

janetkuo commented:
/assign @monopole 

for review & approval on build visibility related change

WanLinghao commented:
/retest

WanLinghao commented:
@soltysh can you plz approve this patch? thanks

janetkuo commented:
@WanLinghao we need approver from **[build/visible_to/OWNERS](https://github.com/kubernetes/kubernetes/blob/master/build/visible_to/OWNERS)**

monopole commented:
/approve
/lgtm

soltysh commented on pkg/kubectl/cmd/rollout/rollout_pause_test.go:
```diff
@@ -0,0 +1,121 @@
+	cmd := NewCmdRolloutPause(tf, buf)
+
+	cmd.Run(cmd, []string{deploymentName})
+	expectedOutput := "deployment.apps \"" + deploymentName + "\" paused\n"
```
@janetkuo @monopole @kargakis please do not merge tests that are ONLY verifying the output of the command without verifying the actual change has been applied. This gives false assumption the command is tested, and adds to the overall tech-debt sig-cli already has, and is currently on a path to clean.

lavalamp replied:
Agree, and I thought we didn't make promises about the output format unless you specify some -o option. So this thing ought not be enforced.

soltysh replied:
That is correct.
